### PR TITLE
chore: use maven artifacts instead of local builds

### PR DESCRIPTION
**Pull request by paullatzelsperger**
## What this PR changes/adds

This PR switches from checkout out and building locally the EDC core code to simply using maven artifacts from MavenCentral.

## Why it does that

To streamline the build process.

## Checklist

- [ ] added appropriate tests?
- [ ] performed checkstyle check locally?
- [ ] added/updated copyright headers?
- [ ] documented public classes/methods?
- [ ] added/updated relevant documentation?
- [ ] added relevant details to the changelog? (_skip with label `no-changelog`_)
- [ ] formatted title correctly? (_take a look at the [CONTRIBUTING](https://github.com/eclipse-dataspaceconnector/RegistrationService/blob/main/CONTRIBUTING.md#submit-a-pull-request) and [styleguide](https://github.com/eclipse-dataspaceconnector/RegistrationService/blob/main/styleguide.md) for details_)

### CODE REVIEW

**Review by algattik — APPROVED** [review]

**Comment by paullatzelsperger** [comment]
could make sense to wait until #18 is merged, because that introduces a dependency onto IdentityHub. We could also adapt that so that the IH Dependency goes against MavenCentral

**Comment by paullatzelsperger** [comment]
will close this, as this needs to be done in a larger, overarching design.

**Pull request closed without merging**
